From 8ca39106b2a8ba4413e81fd36c44d4fdac1ea9a4 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 28 Mar 2023 17:50:06 +0300
Subject: [PATCH] [SW-0] erom: switch to use erom app name

eROM Recovery prints Preboot version string. This is misleading and can
be interpreted wrongly.
Add ability to use different app names in the version string and add
eROM version string.

Change-Id: Id17ab19286c31666db1870193b35b61207dcc8f2
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ext_rom/erom_app_defs.h b/app/ext_rom/erom_app_defs.h
index 67be318..d353d91 100644
--- a/app/ext_rom/erom_app_defs.h
+++ b/app/ext_rom/erom_app_defs.h
@@ -7,5 +7,6 @@
 #define _EROM_APP_DEFS_H_
 
 #define HL_FW_COMP	HL_FW_COMP_EROM
+#define HL_FW_COMP_NAME	"eROM Recovery"
 
 #endif /* _EROM_APP_DEFS_H_ */
diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index cb5e48b..b087b1a 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -115,7 +115,7 @@
 	},
 };
 
-static const char *preboot_ver_str = "Preboot " VERSION_STR;
+static const char *preboot_ver_str = HL_FW_COMP_NAME " " VERSION_STR;
 
 #define BTL_VER_PREFIX	"BTL version "
 #define FUSE_VER_PREFIX	"Fuse version "
diff --git a/app/preboot/preboot_app_defs.h b/app/preboot/preboot_app_defs.h
index e6c2e27..0e09a02 100644
--- a/app/preboot/preboot_app_defs.h
+++ b/app/preboot/preboot_app_defs.h
@@ -7,5 +7,6 @@
 #define _PREBOOT_APP_DEFS_H_
 
 #define HL_FW_COMP	HL_FW_COMP_PREBOOT
+#define HL_FW_COMP_NAME	"Preboot"
 
 #endif /* _PREBOOT_APP_DEFS_H_ */
